File: test/hello_mpi.c
#include <mpi.h>
#include <stdio.h>

int main() {
    MPI_Init(NULL, NULL);
    int world_size;
    MPI_Comm_size(MPI_COMM_WORLD, &world_size);
    int world_rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);

    char proc_name[MPI_MAX_PROCESSOR_NAME];
    int len;
    MPI_Get_processor_name(proc_name, &len);
    printf("hello from MPI in C\n");
    MPI_Finalize();
    return 0;
}

File: test/hello_mpi.f90
program hello
  implicit none
  include 'mpif.h'
  integer rank, size, ierror, tag, status(MPI_STATUS_SIZE)
   
  call MPI_INIT(ierror)
  call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierror)
  call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierror)
  print*, "hello from MPI in Fortran"
  call MPI_FINALIZE(ierror)

end program hello